File: mobile/android/app/src/main/java/com/radiostream/player/PlaylistControls.java
package com.radiostream.player;

import org.jdeferred.Promise;

/**
 * Created by vitaly on 18/11/2016.
 */
public interface PlaylistControls {
    Promise<Song, Exception, Void> play();

    void pause();

    void playPause();

    void playNext();
}

File: app/android/app/src/main/java/com/radiostream/player/PlaylistControls.kt
package com.radiostream.player

import org.jdeferred.Promise

import kotlinx.coroutines.experimental.Job

/**
 * Created by vitaly on 18/11/2016.
 */
interface PlaylistControls {
    suspend fun play()

    fun pause()

    suspend fun playNext()

    suspend fun skipToSongByIndex(index: Int)
}
